beluga/include/beluga/sensor/beam_model.hpp

```diff
@@ -122,42 +122,47 @@ class BeamSensorModel : public Mixin {
   [[nodiscard]] weight_type importance_weight(const state_type& state) const {
     const auto beam = Ray2d{grid_, state, params_.beam_max_range};
-    return std::transform_reduce(points_.cbegin(), points_.cend(), 0.0, std::plus{}, [this, &beam](const auto& point) {
-      // TODO(Ramiro): We're converting from range + bearing to cartesian points in the ROS node, but we want range +
-      // bearing here. We might want to make that conversion in the likelihood model instead, and let the measurement
-      // type be range, bearing instead of x, y.
-
-      // Compute the range according to the measurement.
-      const double z = std::sqrt(point.first * point.first + point.second * point.second);
-
-      // Compute range according to the map (raycasting).
-      const auto beam_bearing = Sophus::SO2d{point.first, point.second};
-      const double z_mean = beam.cast(beam_bearing).value_or(params_.beam_max_range);
-
-      // 1: Correct range with local measurement noise.
-      const double eta_hit = 2. / (std::erf((params_.beam_max_range - z_mean) / (std::sqrt(2.) * params_.sigma_hit)) -
-                                   std::erf(-z_mean / (std::sqrt(2.) * params_.sigma_hit)));
-      const double d = (z - z_mean) / params_.sigma_hit;
-      const double n = 1. / (std::sqrt(2. * M_PI) * params_.sigma_hit);
-      double pz = params_.z_hit * eta_hit * n * std::exp(-(d * d) / 2.);
-
-      // 2: Unexpected objects.
-      if (z < z_mean) {
-        const double eta_short = 1. / (1. - std::exp(-params_.lambda_short * z_mean));
-        pz += params_.z_short * params_.lambda_short * eta_short * std::exp(-params_.lambda_short * z);
-      }
-
-      // 3 and 4: Max range return or random return.
-      if (z < params_.beam_max_range) {
-        pz += params_.z_rand / params_.beam_max_range;
-      } else {
-        pz += params_.z_max;
-      }
-
-      // TODO(glpuga): Investigate why AMCL and QuickMCL both use this formula for the weight.
-      // See https://github.com/Ekumen-OS/beluga/issues/153
-      return pz * pz * pz;
-    });
+    const double n = 1. / (std::sqrt(2. * M_PI) * params_.sigma_hit);
+    return std::transform_reduce(
+        points_.cbegin(), points_.cend(), 0.0, std::plus{}, [this, &beam, n](const auto& point) {
+          // TODO(Ramiro): We're converting from range + bearing to cartesian points in the ROS node, but we want range
+          // + bearing here. We might want to make that conversion in the likelihood model instead, and let the
+          // measurement type be range, bearing instead of x, y.
+
+          // Compute the range according to the measurement.
+          const double z = std::sqrt(point.first * point.first + point.second * point.second);
+
+          // dirty hack to prevent SO2d from calculating the hypot again to normalize the vector.
+          auto beam_bearing = Sophus::SO2d{};
+          beam_bearing.data()[0] = point.first / z;
+          beam_bearing.data()[1] = point.second / z;
```

Comment:
@glpuga meta: these computations are  something that we know we shouldn't be doing, and will go away once https://github.com/Ekumen-OS/beluga/issues/170 is addressed (and the TODO is resolved).